rundeckapp/grails-spa/packages/ui-trellis/src/app/pages/job/browse/JobListScmStatus.vue

```diff
@@ -53,66 +69,64 @@ export default defineComponent({
         defaultDisplayText() {
             if(this.displayExport){
               return this.exportDisplayText
             }else{
               return this.importDisplayText
             }
         },
-        importDisplayText() {
-            switch (this.importState) {
-                case "IMPORT_NEEDED":
-                    return this.$t(
-                        "scm.import.status.IMPORT_NEEDED.display.text"
-                    );
-                case "REFRESH_NEEDED":
-                    return this.$t(
-                        "scm.import.status.REFRESH_NEEDED.display.text"
-                    );
-                case "UNKNOWN":
-                    return this.$t("scm.import.status.UNKNOWN.display.text");
-                case "CLEAN":
-                    return this.$t("scm.import.status.CLEAN.display.text");
-                case "LOADING":
-                    return this.$t("scm.import.status.LOADING.display.text");
+        exportErrorText(){
+            if(this.scmExport?.validationError){
+              return this.scmExport?.validationError
+            }else if (this.scmExport?.validationErrorCode){
+              return this.$t(this.scmExport?.validationErrorCode)
+            }else{
+              return this.$t('scm.export.status.ERROR.display.text')
             }
-            return this.importState;
         },
-        exportDisplayText() {
-            switch (this.exportState) {
-                case "EXPORT_NEEDED":
-                    return this.$t(
-                        "scm.export.status.EXPORT_NEEDED.display.text"
-                    );
-                case "CREATE_NEEDED":
-                    return this.$t(
-                        "scm.export.status.CREATE_NEEDED.display.text"
-                    );
-                case "REFRESH_NEEDED":
-                    return this.$t(
-                        "scm.export.status.REFRESH_NEEDED.display.text"
-                    );
-                case "DELETED":
-                    return this.$t("scm.export.status.DELETED.display.text");
-                case "CLEAN":
-                    return this.$t("scm.export.status.CLEAN.display.text");
-                case "LOADING":
-                    return this.$t("scm.export.status.LOADING.display.text");
+        importErrorText() {
+            if(this.scmImport?.validationError){
+              return this.scmImport?.validationError
+            }else if (this.scmImport?.validationErrorCode){
+              return this.$t(this.scmImport?.validationErrorCode)
+            }else{
+              return this.$t('scm.import.status.ERROR.display.text')
             }
-            return this.exportState;
+        },
+        importDisplayText() {
+          return this.$t(
+            `scm.import.status.${this.importState}.display.text`
+          );
+        },
+        exportDisplayText() {
+          return this.$t(
+            `scm.export.status.${this.exportState}.display.text`
+          );
```

Comment:
tiny nit: when there is an error with import or export, both importDisplayText and exportDisplayText values will include undefined in the string, while this doesn't affect anything in the UI, I'd like to suggest returning an empty string in this case so that if one day the UI is changed to always show these strings, there's no risk of showing values that do not contain translations.


![Screenshot 2024-02-13 at 5 15 08 PM](https://github.com/rundeck/rundeck/assets/139791797/710fa3ca-c0ca-49f5-b7ab-6fe3f6683310)
![Screenshot 2024-02-13 at 5 10 01 PM](https://github.com/rundeck/rundeck/assets/139791797/d33d3eab-6c9f-4deb-b577-ad67567d9b12)

Reply:
Thanks, makes sense to do that